transcom/mymove
# B-19535-INT-KB authorized weight updates to 110% of estimated weights

## [B-19535](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19535)
## [ORIGINAL MAIN PR](https://github.com/transcom/mymove/pull/12670)
## [ORIGINAL INT PR](https://github.com/transcom/mymove/pull/12597)
## [REVERT](https://github.com/transcom/mymove/pull/12703)

## Summary

As estimated weights come in on approved HHGs, the entitlements.authorized_weight (database)/ max billable weight (front end) should be updated to be 110% of the total estimated weight. This can be adjusted up or down by office users.

Original PRs to INT and MAIN had a bug where a shipment could be counted twice if the shipment itself was updated after it already had an estimated weight. This fixes that.

Also includes updates to the MoveDetails page and its children, where we were improperly displaying the authorized weight as the weight allowance.

How to test

~~1. Create a new move with a few HHGs~~
~~2. Approve as SC, then as a TOO approve one HHG.~~
~~3. As PRIME, update a shipment to have an estimated weight~~
~~4. Login as TOO and navigate to move task order page. The Max billable weight should align with the 110% of estimated weight. Click the pencil and adjust the Max billable weight and confirm that the adjustment persists.~~
~~5. As PRIME, create a new shipment with an estimated weight.~~
~~6. Login as TOO and navigate to move task order page. The Max billable weight should align with the 110% of estimated weight.~~

1. Get a move with an HHG or two to a state where it's shipments need to be approved by a TOO
2. The expected behavior is the max billable weight should recalculate to be 110% of the total estimated weight of all non PPM shipments on a move. This value is calculated either when an approved shipment receives an estimated weight, or a shipment is approved. If the office users updates the max billable weight, that value should persist until the value is recalculated or they changes it again
3. Make some combinations of NTS, NTS-R, and HHG shipments and give them estimated weights, or approve them and give them estimated weights and make sure these figures match
4. Try and break it

Note: canceled shipments are not in the scope of this work, as that work needs clarification. A canceled shipment should not recalculate the max billable weight

![image](https://github.com/transcom/mymove/assets/147535220/9a99f3ea-a890-4058-8876-f5979fc3c93a)
110% OF ESTIMATED WEIGHT == MAX BILLABLE WEIGHT
